tests/components/tasmota/test_common.py

```diff
@@ -44,15 +44,15 @@ async def help_test_availability_when_connection_lost(
     assert state.state != STATE_UNAVAILABLE
 
     mqtt_mock.connected = False
-    mqtt_client_mock.on_disconnect(None, None, 0)
+    hass.add_job(mqtt_client_mock.on_disconnect, None, None, 0)
```

Comment:
We're in async context so must use the async api. Use `hass.async_add_executor_job` and await the task returned.

Reply:
Fixed!